src/modules/ipfs/ipfs.module.ts

```diff
@@ -1,38 +1,109 @@
+import { DidStore as DidStoreGateway } from 'didStoreInfura';
+import { DidStore as DidStoreCluster } from 'didStoreCluster';
 import { Module, Global } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
+import { IPFSController } from './ipfs.controller';
 import { IPFSService } from './ipfs.service';
-import { IpfsConfig } from './ipfs.types';
+import {
+  IpfsClusterConfig,
+  IPFSClusterConfigToken,
+  IpfsInfuraConfig,
+  IPFSInfuraConfigToken,
+  PIN_CLAIM_QUEUE_NAME,
+} from './ipfs.types';
+import { BullModule } from '@nestjs/bull';
+import { PinProcessor } from './pin.processor';
 
-const IPFSClientConfigProvider = {
-  provide: 'IPFSClientConfig',
-  useFactory: (config: ConfigService): IpfsConfig => {
-    const IPFS_CLIENT_PORT = config.get<string>('IPFS_CLIENT_PORT');
+const IPFSInfuraConfigProvider = {
+  provide: IPFSInfuraConfigToken,
+  useFactory: (config: ConfigService): IpfsInfuraConfig => {
+    const IPFS_CLIENT_URL = config.get<string>('IPFS_CLIENT_URL');
+    const IPFS_CLIENT_PROTOCOL = config.get<string>('IPFS_CLIENT_PROTOCOL');
     const IPFS_CLIENT_HOST = config.get<string>('IPFS_CLIENT_HOST');
+    const IPFS_CLIENT_PORT = config.get<string>('IPFS_CLIENT_PORT');
+    const ipfsConfig: IpfsInfuraConfig = {
+      url: IPFS_CLIENT_URL,
```

Comment:
Why was `url` introduced in `IpfsInfuraConfig`? It seems a URL could be derived from `protocol`, `host` and `port`, so it seems to me like it might conflict with those values

Reply:
Ah yes, I see from the documentation that `url` supersedes the other values: https://github.com/ipfs/js-ipfs/tree/master/packages/ipfs-http-client#createoptions